File: android/app/src/main/java/com/myvu/client/reminder/ReminderActionReceiver.java
package com.myvu.client.reminder;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.myvu.client.core.LogBus;
import com.myvu.client.database.Reminder;
import com.myvu.client.database.ReminderRepository;

public class ReminderActionReceiver extends BroadcastReceiver {

    public static final String ACTION_COMPLETE = "com.myvu.client.reminder.ACTION_COMPLETE";
    public static final String ACTION_SNOOZE = "com.myvu.client.reminder.ACTION_SNOOZE";

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null || intent.getAction() == null) return;

        long reminderId = intent.getLongExtra("reminder_id", -1);
        if (reminderId == -1) return;

        String action = intent.getAction();
        ReminderRepository repo = new ReminderRepository(context);

        // Cancel Phone Notification
        NotificationManager nm = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
        if (nm != null) {
            nm.cancel((int) reminderId);
        }

        // Dismiss HUD notification
        ReminderNotifier.dismissGlassesNotification(context, reminderId);

        if (ACTION_COMPLETE.equals(action)) {
            LogBus.log("ReminderActionReceiver -> Marking reminder #" + reminderId + " as COMPLETED");
            repo.updateReminderState(reminderId, "COMPLETED");
        } else if (ACTION_SNOOZE.equals(action)) {
            long newTriggerAt = System.currentTimeMillis() + (10 * 60 * 1000); // +10 minutes
            LogBus.log("ReminderActionReceiver -> Snoozing reminder #" + reminderId + " +10m");

            Reminder r = repo.getReminder(reminderId);
            if (r != null && repo.snoozeReminder(reminderId, newTriggerAt)) {
                ReminderScheduler.scheduleReminder(context, reminderId, newTriggerAt, r.getAlarmRequestCode());
            }
        }
    }
}

File: android-kotlin/app/src/main/java/com/myvu/client/reminder/ReminderActionReceiver.kt
package com.myvu.client.reminder

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.myvu.client.core.LogBus
import com.myvu.client.database.ReminderRepository

class ReminderActionReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {
        if (context == null || intent == null || intent.action == null) return

        val reminderId = intent.getLongExtra("reminder_id", -1)
        if (reminderId == -1L) return

        val action = intent.action
        val repo = ReminderRepository(context)

        val nm = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager?
        nm?.cancel(reminderId.toInt())

        ReminderNotifier.dismissGlassesNotification(context, reminderId)

        if (ACTION_COMPLETE == action) {
            LogBus.log("ReminderActionReceiver -> Marking reminder #$reminderId as COMPLETED")
            repo.updateReminderState(reminderId, "COMPLETED")
        } else if (ACTION_SNOOZE == action) {
            val newTriggerAt = System.currentTimeMillis() + (10 * 60 * 1000)
            LogBus.log("ReminderActionReceiver -> Snoozing reminder #$reminderId +10m")

            val r = repo.getReminder(reminderId)
            if (r != null && repo.snoozeReminder(reminderId, newTriggerAt)) {
                ReminderScheduler.scheduleReminder(context, reminderId, newTriggerAt, r.alarmRequestCode)
            }
        }
    }

    companion object {
        const val ACTION_COMPLETE = "com.myvu.client.reminder.ACTION_COMPLETE"
        const val ACTION_SNOOZE = "com.myvu.client.reminder.ACTION_SNOOZE"
    }
}
